src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2170,7 +2171,24 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromAliases(const Identifier
         auto & alias_identifier_node = it->second->as<IdentifierNode &>();
         auto identifier = alias_identifier_node.getIdentifier();
         auto lookup_result = tryResolveIdentifier(IdentifierLookup{identifier, identifier_lookup.lookup_context}, scope, identifier_resolve_settings);
-        it->second = lookup_result.resolved_identifier;
+        if (lookup_result.isResolved())
```

Comment:
If we do not resolve identifier node that has alias, and we already bind to alias, we need to throw exception. This works the same way for tables. No need to remove anything from map, just throw exception.
I suspect that I added some unnecessary checks here. It is better to throw exception, and check if it breaks something.